### PR TITLE
fix(core): replace binjumper with cmd script without prompt

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

Some users are running in protected systems where running unknown binaries is not allowed.

Fixes https://github.com/yarnpkg/berry/issues/1938

**How did you fix it?**

Revert https://github.com/yarnpkg/berry/pull/1808 and use cmd again but with a workaround to get rid of the `Terminate batch job` prompt as described in https://github.com/microsoft/terminal/issues/217#issuecomment-737594785 by @rivy

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by merceyz** [comment]
@markerikson Could you check if this works for you? `yarn set version from sources --branch 1939`, it includes the changes from #1934

**Comment by arcanis** [comment]
That will work from scripts, but not if it calls a JS script that calls `spawn` itself, right? Since it won't get called by powershell.

**Comment by merceyz** [comment]
Correct, it only gets rid of the prompt when the bin is spawned by us. Plan on adding a config if @markerikson confirms this works, unsure if the native bin should be a opt-in or opt-out.

**Comment by arcanis** [comment]
> Correct, it only gets rid of the prompt when the bin is spawned by us

I'd tend to prefer the setting to fallback on the classic legacy behavior (cmd scripts), not on powershell. If one magic isn't enough, I'm reticent about including a second one (ie locating the powershell binary etc) that has its own drawbacks.

**Comment by markerikson** [comment]
I can confirm that this _attempts_ to execute things via Powershell.  Unfortunately, the Bit9 security software still flags this as a "suspicious pattern".  

It's likely this will work fine for everyone else, and I can try to report this to our internal IT center to see if they can tweak the settings to allow it, but unfortunately it doesn't work for me personally atm.

**Comment by merceyz** [comment]
That's what I was afraid of, I've reverted the powershell change and disabled the native binjumper so it should be as it was before now. Mind trying again?

**Comment by markerikson** [comment]
Hey, that actually looks like it worked!

Haven't run any of the app yet, but:

```
➤ YN0000: ┌ Link step
➤ YN0062: │ fsevents@patch:fsevents@npm%3A2.1.2#builtin<compat/fsevents>::version=2.1.2&hash=127e8e The platform win32 is incompatible with this module, building skipped.
➤ YN0062: │ fsevents@patch:fsevents@npm%3A1.2.12#builtin<compat/fsevents>::version=1.2.12&hash=127e8e The platform win32 is incompatible with this module, building skipped.
➤ YN0007: │ nodemailer@npm:6.4.6 must be built because it never did before or the last one failed
➤ YN0007: │ core-js@npm:2.6.11 must be built because it never did before or the last one failed
➤ YN0007: │ core-js-pure@npm:3.6.4 must be built because it never did before or the last one failed
➤ YN0007: │ core-js@npm:3.6.4 must be built because it never did before or the last one failed
➤ YN0000: └ Completed in 14s 498ms
➤ YN0000: Done with warnings in 17s 673ms
```

and no security software popup warnings this time.

**Comment by merceyz** [comment]
Perfect, if you need any help with anything feel free to pop by on discord - https://discord.gg/yarnpkg

**Comment by merceyz** [comment]
Hey @markerikson :wave: 
I'm curious if it was just the `node.exe` file that got blocked because it was unsigned, so I've made yarn use a hardlink for it and the "normal" flow for everything else, could you try this PR again please?

**Comment by markerikson** [comment]
Hey. I've had to set aside playing around with Yarn v2 for the moment.  It's still something I want to get back to, but there's other priorities right now.

**Comment by merceyz** [comment]
Totally understandable :)

File: packages/yarnpkg-core/sources/scriptUtils.ts [hunk 1]
```diff
@@ -30,11 +29,9 @@ enum PackageManager {
 
 async function makePathWrapper(location: PortablePath, name: Filename, argv0: NativePath, args: Array<string> = []) {
   if (process.platform === `win32`) {
-    await Promise.all([
-      xfs.writeFilePromise(ppath.format({dir: location, name, ext: `.exe`}), getBinjumper()),
-      xfs.writeFilePromise(ppath.format({dir: location, name, ext: `.exe.info`}), [argv0, ...args].join(`\n`)),
-      xfs.writeFilePromise(ppath.format({dir: location, name, ext: `.cmd`}), `@"${argv0}" ${args.map(arg => `"${arg.replace(`"`, `""`)}"`).join(` `)} %*\n`),
-    ]);
+    // https://github.com/microsoft/terminal/issues/217#issuecomment-737594785
+    const cmdScript = `@echo off\ngoto #_undefined_# 2>NUL || title %COMSPEC% & "${argv0}" ${args.map(arg => `"${arg.replace(`"`, `""`)}"`).join(` `)} %*`;
```
Comment (rivy) [inline packages/yarnpkg-core/sources/scriptUtils.ts:33]: `@echo off` changes the caller's echo status. Using leading `@` for the commands is more "hygenic".

```suggestion
    const cmdScript = `@goto #_undefined_# 2>NUL || @title %COMSPEC% & @"${argv0}" ${args.map(arg => `"${arg.replace(`"`, `""`)}"`).join(` `)} %*`;
```

Comment (arcanis) [inline packages/yarnpkg-core/sources/scriptUtils.ts]: Do you see a way to test it? I could see that being very difficult, so not a blocker, but preventing regressions would be nice.

Comment (merceyz) [reply packages/yarnpkg-core/sources/scriptUtils.ts]: Afraid not

**Pull request merged**